File: c/0001_two_sum.c
/**
 * Note: The returned array must be malloced, assume caller calls free().
 */

int* twoSum(int* nums, int numsSize, int target, int* returnSize) {
  int *output = malloc(2 * sizeof(int));

  for (int i = 0; i < numsSize; i++) {
    for (int j = i + 1; j < numsSize; j++) {
      if (nums[i] + nums[j] == target) {
        *returnSize = 2;
        output[0] = i;
        output[1] = j;
        
        return output;
      }
    }
  }

  *returnSize = 0;

  return NULL;
}

File: rust/0001_two_sum.rs
impl Solution {
  pub fn two_sum(nums: Vec<i32>, target: i32) -> Vec<i32> {
    for (i, &a) in nums.iter().enumerate() {
      for (j, &b) in nums.iter().enumerate() {
        if i != j && a + b == target {
          return vec![i as i32, j as i32];
        }
      }
    }

    return vec![];
  }
}
